SwissTransferFeatures/NewTransferView/NewTransferManager.swift

```diff
@@ -79,21 +79,34 @@ class NewTransferManager: ObservableObject {
 extension NewTransferManager {
     /// Move the imported files/folder in the temporary directory
     private func moveToTmp(files: [URL]) {
-        do {
-            let tmpDirectory = try URL.tmpUploadDirectory()
-            for file in files {
-                do {
-                    let destination = tmpDirectory.appending(path: file.lastPathComponent)
-                    _ = file.startAccessingSecurityScopedResource()
-                    try FileManager.default.copyItem(at: file, to: destination)
-                } catch {
-                    Logger.general.error("An error occured while copying files: \(error)")
-                }
-                file.stopAccessingSecurityScopedResource()
+        for file in files {
+            do {
+                let destination = try destinationURLFor(source: file)
+                _ = file.startAccessingSecurityScopedResource()
+                try FileManager.default.copyItem(at: file, to: destination)
+            } catch {
+                Logger.general.error("An error occured while copying files: \(error)")
+            }
+            file.stopAccessingSecurityScopedResource()
+        }
+    }
+
+    /// Find a valid name if a file/folder already exist with the same name
+    private func destinationURLFor(source: URL) throws -> URL {
+        let allFiles = try FileManager.default.contentsOfDirectory(at: URL.tmpUploadDirectory(), includingPropertiesForKeys: nil)
+            .map(\.lastPathComponent)
+
+        let shortName = source.deletingPathExtension().lastPathComponent
+        var increment = 0
+        var testName = source.lastPathComponent
+        while allFiles.contains(where: { $0 == testName }) {
```

Comment:
Maybe we can abstract this a bit in order to test it ?

Also we could put all the files of a `Transfer` in a folder (like a namespace) to reduce risks of a race conditon later with a more complex app.